src/main/java/org/fcrepo/upgrade/utils/F47ToF5UpgradeManager.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.upgrade.utils;
+
+import static org.fcrepo.upgrade.utils.HttpConstants.CONTENT_TYPE_HEADER;
+import static org.fcrepo.upgrade.utils.HttpConstants.LOCATION_HEADER;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.FileUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.fcrepo.client.FcrepoLink;
+
+/**
+ * @author dbernstein
+ * @since 2019-08-05
+ */
+class F47ToF5UpgradeManager extends UpgradeManagerBase implements UpgradeManager {
+
+    private static final org.slf4j.Logger LOGGER = getLogger(F47ToF5UpgradeManager.class);
+    private static final Pattern MESSAGE_EXTERNAL_BODY_URL_PATTERN = Pattern.compile("^.*url=\"(.*)\".*$");
+
+    /**
+     * Constructor
+     *
+     * @param config the upgrade configuration
+     */
+    F47ToF5UpgradeManager(final Config config) {
+        super(config);
+    }
+
+    @Override
+    public void start() {
+        //walk the directory structure
+        processDirectory(this.config.getInputDir());
+    }
+
+    private void processDirectory(final File dir) {
+        LOGGER.info("Processing directory: {}", dir.getAbsolutePath());
+        try (final Stream<Path> walk = Files.walk(dir.toPath())) {
+            for (Path path : walk.filter(path -> Files.isRegularFile(path)).collect(Collectors.toList())) {
+                processFile(path);
+            }
```

Comment:
There's not really a reason to collect the paths into a list here, it can keep going with the streaming API, which should be more memory efficient for large directories:
```suggestion
            walk.filter(Files::isRegularFile).forEach(this::processFile);
```